src/drive/web/modules/move/MoveModal.jsx

```diff
@@ -1,67 +1,87 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import { compose } from 'redux'
 import { Modal } from 'cozy-ui/react'
-import { Query } from 'cozy-client'
+import { translate } from 'cozy-ui/react/I18n'
+import { Query, cancelable, withClient } from 'cozy-client'
 import { ROOT_DIR_ID, TRASH_DIR_ID } from 'drive/constants/config'
 import Alerter from 'cozy-ui/react/Alerter'
 import { connect } from 'react-redux'
 import { getTracker } from 'cozy-ui/react/helpers/tracker'
-import logger from 'lib/logger'
 
-import Header from './Header'
-import Explorer from './Explorer'
-import FileList from './FileList'
-import Loader from './Loader'
-import LoadMore from './LoadMore'
-import Footer from './Footer'
-import Topbar from './Topbar'
-
-class MoveModal extends React.Component {
-  static contextTypes = {
-    client: PropTypes.object.isRequired,
-    t: PropTypes.func.isRequired
-  }
+import { CozyFile } from 'models'
+import logger from 'lib/logger'
+import withSharingState from 'sharing/hoc/withSharingState'
+import Header from 'drive/web/modules/move/Header'
+import Explorer from 'drive/web/modules/move/Explorer'
+import FileList from 'drive/web/modules/move/FileList'
+import Loader from 'drive/web/modules/move/Loader'
+import LoadMore from 'drive/web/modules/move/LoadMore'
+import Footer from 'drive/web/modules/move/Footer'
+import Topbar from 'drive/web/modules/move/Topbar'
+
+export class MoveModal extends React.Component {
   constructor(props) {
     super(props)
-
+    this.promises = []
     const { displayedFolder } = props
     this.state = {
       folderId: displayedFolder ? displayedFolder._id : ROOT_DIR_ID,
       isMoveInProgress: false
     }
   }
 
+  componentWillUnmount() {
+    this.promises.forEach(p => p.cancel())
+  }
+
   navigateTo = folder => {
     this.setState({ folderId: folder.id })
   }
 
+  registerCancelable = promise => {
+    const cancelableP = cancelable(promise)
+    this.promises.push(cancelableP)
+    return cancelableP
+  }
```

Comment:
to follow @ptbrowne advice we should have something like : 

```jsx 
registerCancelable = promise => {
if(!this.promises) this.promises = []
const cancelableP = cancelable(promise)
this.promises.push(cancelableP)
return cancelableP
}
```

```jsx 
unregisterCancelable(){
this.promises.forEach(p => p.cancel())
this.promises = [] 
}
```

Reply:
not sure it applies here since we are cancelling the promises and not really unregistering them. But I agree that a separate method `cancelAllPromises` would be nice

Reply:
In fact, we have to unregister them. Since the contructor is not called everytime, we should be sure to reset our promise array on unmount